src/global/components/Navbar.tsx

```diff
@@ -0,0 +1,91 @@
+import { useLayoutEffect } from "react";
+import { NavLink } from "react-router-dom";
+import { useRecoilState, useRecoilValue } from "recoil";
+import { menuState, mobileState } from "../state/navbarState";
+import MenuDropdown from "./MenuDropdown";
+import { pathSelector } from "../state/navigationState";
+
+export default function Navbar() {
+  const [isMenuOpen, setMenuOpen] = useRecoilState(menuState);
+  const filteredPathsArray = useRecoilValue(pathSelector);
+  const [isMobileSize, setIsMobileSize] = useRecoilState(mobileState);
+
+  const handleOpenMenu = (e: React.MouseEvent<HTMLButtonElement>) => {
+    setMenuOpen(!isMenuOpen);
+  };
+
+  useLayoutEffect(() => {
+    if (window.innerWidth < 768) setIsMobileSize(true);
+  });
+
+  window.addEventListener("resize", () => {
+    if (window.innerWidth < 768 && !isMobileSize) setIsMobileSize(true);
+    if (window.innerWidth >= 768 && isMobileSize) setIsMobileSize(false);
+  });
+
+  return (
+    <nav
+      className={`relative h-navbar-full-height w-screen bg-navbar-dark-gray-color flex items-center !z-10 ${
+        isMenuOpen && isMobileSize
+          ? "animate-navbarMobileColorAnimation"
+          : !isMenuOpen && isMobileSize
+          ? "animate-navbarMobileColorAnimationReverse"
+          : ""
+      }`}
```

Comment:
Check [clsx](https://www.npmjs.com/package/clsx) and [tailwind-merge](https://www.npmjs.com/package/tailwind-merge) for dynamic styling

Reply:
Also if you don't want to return something you can just write `!isMenuOpen && isMobileSize && "styles"`

Reply:
Added separate components (MenuDropdown and MenuDropdownMobile) for different screen sizes to avoid isMobileSize and added isMenuOpen directly into separate components.